modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java

```diff
@@ -380,6 +394,25 @@ private void interruptAndWait(Thread t, long timeout) {
         }
     }
 
+    private void shutdownAndAwaitTermination(ExecutorService pool) {
+        pool.shutdown(); // Disable new tasks from being submitted
+        try {
+            // Wait a while for existing tasks to terminate
+            if (!pool.awaitTermination(60, TimeUnit.SECONDS)) {
+                pool.shutdownNow(); // Cancel currently executing tasks
+                // Wait a while for tasks to respond to being cancelled
+                if (!pool.awaitTermination(60, TimeUnit.SECONDS)) {
+                    log.error("Pool did not terminate");
```

Comment:
Should we throw the exception in this if condition?

Reply:
Would an exception provide any additional information? I can add a stack trace to the log if you think it's useful?

Reply:
To throw the exception is not necessary, it mainly try to interrupt the thread. If the kernel thread stuck, we can dump the process threads info.